### PR TITLE
parser: add opt_rule_blacklist and `admin reload` it.

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve? 
Refer to: https://github.com/pingcap/tidb/pull/11096

### What is changed and how it works?
Just add an `admin reload` in the parser and a new word in the ast.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported variable/fields change

Related changes

 - None

### CODE REVIEW

File: misc.go [hunk 1]
```diff
@@ -589,6 +589,7 @@ var tokenMap = map[string]int{
 	"BINDING":                  binding,
 	"BINDINGS":                 bindings,
 	"EXPR_PUSHDOWN_BLACKLIST":  exprPushdownBlacklist,
+	"DISABLED_OPTIMIZE_LIST":   disabledOptimizeList,
```
Comment (zz-jason) [inline misc.go]: Better to keep the same naming pattern. How about `OPT_RULE_BLACKLIST`?

Comment (lzmhhh123) [reply misc.go]: It contains expression push down and other things like cascade rules in the future besides logical rules. Shall we also change the name?

Comment (zz-jason) [reply misc.go]: I prefer to let this table to only contain the disabled rules. This the PR can be cherry-picked to release-3.0 and don't need to worry about the compatibility problems.

Comment (lzmhhh123) [reply misc.go]: OK, if that's so. I should change it.

**Pull request merged**